networkx/algorithms/matching.py

```diff
@@ -92,13 +98,21 @@ def is_matching(G, matching):
     """
     if isinstance(matching, dict):
         matching = matching_dict_to_set(matching)
+    for e in matching:
+        if len(e) == 1:  # self-loop
+            if next(iter(e)) not in G:
+                msg = f"Matching includes edge {e} not in G"
+                raise nx.NetworkXNotImplemented(msg)
+            continue
+        if not G.has_edge(*e):
+            msg = f"Matching includes edge {e} not in G"
+            raise nx.NetworkXNotImplemented(msg)
     # TODO This is parallelizable.
     return all(len(set(e1) & set(e2)) == 0 for e1, e2 in combinations(matching, 2))
 
 
 def is_maximal_matching(G, matching):
-    """Decides whether the given set or dictionary represents a valid
-    maximal matching in ``G``.
+    """Return True if ``matching`` is a mximal matching of ``G``
```

Comment:
```suggestion
    """Return True if ``matching`` is a maximal matching of ``G``
```

Reply:
I think I'm getting confused between double ` and single `.  
Aren't we supposed to use `G` instead of ``G``?
What's the difference?

Reply:
In principle, parameters are supposed to be in single backticks (so this *should* be `` `G` ``, not ``` ``G`` ```) but in practice it's not a huge difference, so I try not to be *too* picky about it.

It's actually tricky in networkx's case because the parameters-in-single-backticks rule comes from the numpydoc standard, but the `default_role=obj` in the networkx docs configuration actually clashes with this suggestion and results in many more "broken links" when sphinx is run in nitpicky mode. The end result in the rendered html looks good though and I'm not sure the best way to tackle this. So in the mean time my feeling is not to worry about it too much.

Reply:
We changed the default role before. I think it was math and we made it python object. We might have changed it once before that too.  It would be good to have that figured out, but I agree that we shouldn't probably worry about it too much. :}